src/dbus_api.rs

```diff
@@ -0,0 +1,328 @@
+use dbus::Connection as DBusConnection;
+use dbus::{BusType, Path, ConnPath, Message};
+use dbus::arg::{Variant, Iter, Array, Get, RefArg};
+use dbus::stdintf::OrgFreedesktopDBusProperties;
+use dbus::Error;
+
+
+pub const DEFAULT_TIMEOUT: u64 = 10;
+pub const RETRIES_ALLOWED: usize = 10;
```

Comment:
The retries were introduced in ca55a3e, so that we can handle some of the NetworkManager service instabilities and this is the reason I kept the retry count internal.